tests/test_config.py

```diff
@@ -2,47 +2,69 @@
 import logging
 import os
 import socket
+import sys
+import typing
 from copy import deepcopy
+from pathlib import Path
+from types import TracebackType
+from unittest.mock import MagicMock
+
+if sys.version_info < (3, 8):
+    from typing_extensions import Literal
+else:
+    from typing import Literal
 
 import pytest
 import yaml
+from asgiref.typing import ASGIApplication, ASGIReceiveCallable, ASGISendCallable, Scope
+from pytest_mock import MockerFixture
 
 from uvicorn.config import LOGGING_CONFIG, Config
 from uvicorn.middleware.debug import DebugMiddleware
 from uvicorn.middleware.proxy_headers import ProxyHeadersMiddleware
 from uvicorn.middleware.wsgi import WSGIMiddleware
 from uvicorn.protocols.http.h11_impl import H11Protocol
 
+Environ = typing.MutableMapping[str, typing.Any]
+ExcInfo = typing.Tuple[
+    typing.Type[BaseException], BaseException, typing.Optional[TracebackType]
+]
+StartResponse = typing.Callable[
+    [str, typing.Iterable[typing.Tuple[str, str]], typing.Optional[ExcInfo]], None
+]
+
```

Comment:
let's move this to _types.py so that we can use it for the wsgi middleware in the future, with a comment stating it's wsgi stuff

Reply:
Good idea, we can continue adding custom types there. (f9f509b)